src/botPage/bot/Interpreter.js

```diff
@@ -3,9 +3,11 @@ import { observer as globalObserver } from 'binary-common-utils/lib/observer';
 import { createScope } from './CliTools';
 import Interface from './Interface';
 
+const unrecoverableErrors = ['InsufficientBalance', 'CustomLimitsReached'];
 const botInitialized = bot => bot && bot.tradeEngine.options;
 const botStarted = bot => botInitialized(bot) && bot.tradeEngine.tradeOptions;
-const shouldRestartOnError = bot => botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
+const shouldRestartOnError = (bot, e = {}) =>
+    unrecoverableErrors.includes(e.name) && botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
```

Comment:
I think this should be `!unrecoverableErrors.includes(e.name) `

Reply:
If the error is matches the unrecoverableErrors array then it should stop. I have tested for both positive and negative test cases, it seems to be working.

Reply:
But the opposite makes sense to me: If the error is _not_ in `unrecoverableErrors` and bot is initialized and ... then we should restart.